### ISSUE TITLE
Update Checkr API

### ISSUE THREAD

**Issue by dboyer** [Iridescent-CM/technovation-app#3793]
Documentation on the needed Checkr upgrades [is here](https://info.checkr.com/rs/204-WFI-586/images/Report%20Lifecycle%20Design%20Decisions%20Checklist.pdf?mkt_tok=MjA0LVdGSS01ODYAAAGJX_fxiOEq5w7A4Ozg0bzEJBj_tLFqTkH0YNxKF8IY_Fg996gN2QO3okur8WBb_V8fgeDSITKsO-j_qUn32iAjGMTiW2BIHQsmvRwHBdvbZw).

Specifically, we need to:

1. Add the "includes_canceled" parameter to our Checkr integration
2. Add an option for the canceled status anywhere we display a check's status
3. Add results and status parameters to any reports

We can split out a ticket covering some initial investigation into our Checkr API if that is helpful.

## QA STEPS

- Pick a candidate from this spreadsheet (one that doesn't have a middle name):
https://docs.google.com/spreadsheets/d/11ZeECuXArp7x3DoBXW-pz-Lgbe5uHvpCGGK1ZrdOMkU/edit#gid=0
- Log in as any mentor from the US and input the details from the spreadsheet onto the mentor's background check page (shown below):
<img src="https://user-images.githubusercontent.com/58957909/224445283-567d04f4-58e2-4749-b543-db7145243fce.png" width="600">

- After the background check has been submitted, check the status by clicking on "Background Check" in the sidebar, then click the green "Check Submission Status" link
- Verify the final background check status (clear or consider) is showing up on the background check status page
  - This background check status page needs to be refreshed manually to update the background check status

If a candidate comes back as "consider", log in to our Checkr staging account (the credentials are in the password safe), click on the candidate, and click the big green "Engage" button (shown in the screenshot below), then back on the TG platform, refresh the status and they should be "clear".
<img src="https://user-images.githubusercontent.com/58957909/224447278-d69e3030-1161-4426-9f11-cacf4cde6ff0.png" width="600">

**Comment by dboyer** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Technovation-605271851956b1000ec2f24d/issues/Iridescent-CM/technovation-app/3793?planning-poker) with Zenhub @shaunxp20 @viviancan

**Comment by shaunxp20** [comment]
One last thing I'm trying to get resolved (before code review and putting it on QA) is to get the hierarchy feature disabled for our staging account.

When creating/requesting a new background report, `work_locations` is required for hierarchy accounts, via their [documentation](https://docs.checkr.com/#operation/createReport):

> work_locations: Array of objects (WorkLocation) Required for hierarchy-enabled accounts.

And, also via their [documentation](https://docs.checkr.com/#tag/Hierarchy-(Enabled-by-request)), it says hierarchies are enabled by request, it must have been enabled by default for our staging account.

It's an issue b/c I had to add this extra `work_locations` field when I was testing locally; it only works when this field is present. But this field isn't currently being used for our production Checkr account (it would require more work/testing to incorporate this field).

I sent out an email to try to get this resolved.

TLDR; trying to get our staging Checkr environment to match up w/ production (so the `work_locations` field isn't required).

**Comment by dboyer** [comment]
Thanks for getting everything to line up correctly! That will be helpful for our next round of work with Checkr.

**Comment by shaunxp20** [comment]
A few notes as I'm wrapping this one up...

The two main changes we had to make were:
- Adding support for the `result` parameter in the Reports payload
- Adding support for the "canceled" value for `status`

I tested the `result` parameter using Checkr's test candidates, afaict this is working as expected, but I couldn't get a candidate into the canceled status, I think my change for this is a pretty safe change though and *should work as expected, also, I don't think this is going to be a common status being returned and I'm guessing we might not see it or have to worry about it for a while.

**Comment by shaunxp20** [comment]
Another thing that I think I mentioned very early on is that we have an admin endpoint to view background checks, it can be viewed on QA or Production, but appending `admin/background_checks` to either base URL.

One piece of additional work would be to remove this code, or expose it in the admin and clean it up/improve it a bit.

**Comment by shaunxp20** [comment]
Moving this one back to "In Progress". I'm running into an issue on QA where a candidate got into the "consider" state (this didn't happen for me when I was testing locally), and now they're stuck in "consider". I need to make some updates to address/fix this.

**Comment by shaunxp20** [comment]
> Moving this one back to "In Progress". I'm running into an issue on QA where a candidate got into the "consider" state (this didn't happen for me when I was testing locally), and now they're stuck in "consider". I need to make some updates to address/fix this.

This issue has been fixed/resolved on QA now. 😄

**Comment by shaunxp20** [comment]
Just documenting this confusing piece of code that I ran into that I didn't understand at first:

https://github.com/Iridescent-CM/technovation-app/blob/99070b11eec6c587412ab9a2d341b258c6089e2f/app/technovation/background_checking.rb#L12

If a background report comes back as "consider", the way it becomes "clear" is by someone logging into Checkr and clicking the "Engage" button (shown in the screenshot below), this is the "engaged" referenced in the code above.

<img src="https://user-images.githubusercontent.com/58957909/224448106-8a797ddd-9f7d-4bc3-a534-8a7435f9a250.png" width="600">
